autogpt_platform/backend/backend/server/v2/store/db.py

```diff
@@ -827,36 +886,353 @@ async def review_store_submission(
             else prisma.enums.SubmissionStatus.REJECTED
         )
 
-        update_data: prisma.types.StoreListingSubmissionUpdateInput = {
-            "Status": submission_status,
-            "reviewComments": comments,
-            "Reviewer": {"connect": {"id": reviewer_id}},
-            "StoreListing": {"connect": {"id": store_listing_version.StoreListing.id}},
-        }
-
-        create_data: prisma.types.StoreListingSubmissionCreateInput = {
-            **update_data,
-            "StoreListingVersion": {"connect": {"id": store_listing_version_id}},
-        }
+        # Update the version with review information
+        update_data = prisma.types.StoreListingVersionUpdateInput(
+            submissionStatus=submission_status,
+            reviewComments=external_comments,
+            internalComments=internal_comments,
+            reviewer={"connect": {"id": reviewer_id}},
+            storeListing={"connect": {"id": store_listing_version.storeListing.id}},
+            reviewedAt=datetime.now(),
+        )
 
-        submission = await prisma.models.StoreListingSubmission.prisma().upsert(
-            where={"storeListingVersionId": store_listing_version_id},
-            data={
-                "create": create_data,
-                "update": update_data,
-            },
+        # Update the version
+        submission = await prisma.models.StoreListingVersion.prisma().update(
+            where={"id": store_listing_version_id},
+            data=update_data,
+            include={"storeListing": True},
         )
 
         if not submission:
-            raise fastapi.HTTPException(  # FIXME: don't return HTTP exceptions here
-                status_code=404,
-                detail=f"Store listing submission {store_listing_version_id} not found",
+            raise backend.server.v2.store.exceptions.DatabaseError(
+                f"Failed to update store listing version {store_listing_version_id}"
             )
 
-        return submission
+        # Convert to Pydantic model for consistency
+        return backend.server.v2.store.model.StoreSubmission(
```

Comment:
better to make this a static `from_db(Submission)` like method in the class so we can re-use it